plugins/wasm-bpf/bpf-api.h

```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2023, eunomia-bpf
+ * All rights reserved.
+ */
+// SPDX-License-Identifier: (LGPL-2.1 OR BSD-2-Clause)
+/* Copyright (c) 2022 Hengqi Chen */
+
+#ifndef __BPF_WASM_API_H
+#define __BPF_WASM_API_H
+
+#include "executor/executor.h"
+#include "runtime/instance/module.h"
+#include "wasmedge/wasmedge.h"
+#include <cinttypes>
+#include <cstdlib>
+#include <memory>
+#include <unordered_set>
+#include <vector>
+extern "C" {
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+}
+
+#define POLL_TIMEOUT_MS 100
+#define DEBUG_LIBBPF_RUNTIME 0
+
+#define PERF_BUFFER_PAGES 64
+
+typedef int (*bpf_buffer_sample_fn)(void *ctx, void *data, size_t size);
+/// An absraction of a bpf ring buffer or perf buffer from bcc.
```

Comment:
Do not use any code from bcc now. Please write our own code like https://github.com/eunomia-bpf/wasm-bpf/blob/225207c068d53bede2ae94d9142bc9430e1fdfe7/runtime/cpp/src/wasm-bpf.cpp#L73